application-mocca-calendar-ui/src/main/resources/MoccaCalendar/MoccaCalendarEventSheet.xml

```diff
@@ -256,6 +293,21 @@ $doc.display($field)
     : $doc.display('allDay')
     ; &lt;label for="description"&gt;$escapetool.xml($doc.displayPrettyName('description', false, false))&lt;/label&gt;
     : $eventInstance.getDescriptionHtml()
+      ## The colors here are for the XWiki Object (from the page url).
+      ## Colors displayed in the calendar are defined in the MoccaCalendarScriptService.java file.
+      ## There is another validation check made in MoccaCalendarScriptService.java. See the fillInColorsFromNearestCalendar() method.
+    ; &lt;label for="backgroundColor"&gt;$escapetool.xml($doc.displayPrettyName('backgroundColor', false, false))&lt;/label&gt;
+    #if ($doc.getValue('backgroundColor').length() == 0) ## If the color is valid or should default to parent calendar colors.
```

Comment:
Aren't you already defaulting on the calendar value in case this is not defined? Why you need to do this again in here?

Reply:
This is only for the View page, which is not directly part of the calendar (it can be viewed as a standalone page).

In the case when the color is not set, the color displayed in the UI would be black. This #if makes it so the color from the parent calendar is displayed instead, with the "(Calendar Default)" text. For invalid colors (like "z") the color displayed in the View page is still black.

The css in the calendar macro normally overrides the css of the event (if the color is invalid) so the color is a prettier default (blue and white instead of black and black). 

View page without if:
![image](https://github.com/user-attachments/assets/aabc1d38-39d9-4a4d-9a65-ecf6ae1a979a)
View page with if:
![image](https://github.com/user-attachments/assets/c196ef78-7422-4644-b6d8-1521fed5df81)

Reply:
You should simply use `$eventInstance.getTextColor()/getBackgroundColor()`, which will give you the right color (from event or parent calendar). I find it ok even if it means not displaying the `Calendar default` info (but could also be done)
Practically you would have something similar to: 
```
#set ($textColor = $eventInstance.getTextColor())
#set (isCalendarDefault = ($textColor == $calendarDoc.getValue('color')) )
```
And then use the `$isCalendarDefault` variable in order to display or not the `(Calendar default)` message
Not that this message should be added using a [translation key](https://extensions.xwiki.org/xwiki/bin/view/Extension/Localization/)